### PR TITLE
feat(tree-item): add tree item component

### PR DESCRIPTION
**Pull request by ricardogoncalves89**
**NIGHT MODE:**
>imageimage

**LIGHT MODE:**
>imageimage

### CODE REVIEW

**Review by GoncaloCamaz — APPROVED** [review]

File: packages/react-ui-components/src/stories/Components/TreeItem.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
+import { EIconName, EOtherIconName } from '@kelvininc/ui-components';
+import { ComponentStory } from '@storybook/react';
+import React from 'react';
+import { KvTreeItem, ETreeItemStates } from '../../components';
+
+// Required to have the correct TagName in the code sample
+KvTreeItem.displayName = 'KvTreeItem';
+
+export default {
+	title: 'Navigation/Tree/Tree Item',
```
Comment (pedrorebelocosta) [inline packages/react-ui-components/src/stories/Components/TreeItem.stories.tsx:10]: Wouldn't this component be a better fit in the Data Display category? I understand the consideration regarding Navigation, but I'm unsure if it's the best category for it.

**Review by pedrorebelocosta — APPROVED** [review]
just a single consideration :P

File: packages/react-ui-components/src/stories/Components/TreeItem.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+			control: 'select',
+			options: Object.values(ETreeItemStates)
+		}
+	},
+	parameters: {
+		notes: require('@ui-notes/tree-item/readme.md')
+	}
+};
+
+const handleToggleExpand = (e: any) => console.log('onToggleExpand', e);
```
Comment (miguelpinto98) [inline packages/react-ui-components/src/stories/Components/TreeItem.stories.tsx:31]: We should move the actions callbacks to `argTypes` with this options:

```
onItemClick: {
		action: 'clicked'
},
onToggleExpand: {
		action: 'clicked'
}
```

After that we can remove from the template the bind for this function.

File: packages/react-ui-components/src/stories/Components/TreeItem.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+WithCounterState.args = {
+	label: 'Node title',
+	additionalLabel: 'default state node',
+	counter: 32,
+	counterState: ETreeItemStates.Success,
+	onItemClick: handleItemClick,
+	onToggleExpand: handleToggleExpand
+};
+
+const ChildrenTreeTemplate: ComponentStory<typeof KvTreeItem> = args => {
```
Comment (miguelpinto98) [inline packages/react-ui-components/src/stories/Components/TreeItem.stories.tsx:109]: `args` isn't used and the `counter` receive a number by definition, this are generating some warnings

**Review by miguelpinto98 — CHANGES_REQUESTED** [review]
Seems very good 💪  
Only a few small changes to go

File: packages/ui-components/src/components/tree-item/tree-item.types.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export enum ETreeItemStates {
```
Comment (miguelpinto98) [inline packages/ui-components/src/components/tree-item/tree-item.types.ts:1]: Use singular for enums `ETreeItemState`

**Review by miguelpinto98 — APPROVED** [review]
👊

**Pull request merged**